impeller/entity/geometry/ellipse_geometry.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/entity/geometry/ellipse_geometry.h"
+
+#include "flutter/impeller/tessellator/circle_tessellator.h"
+
+namespace impeller {
+
+EllipseGeometry::EllipseGeometry(Point center, Scalar radius)
+    : center_(center), radius_(radius) {
+  FML_DCHECK(radius >= 0);
+}
+
+GeometryResult EllipseGeometry::GetPositionBuffer(
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) const {
+  auto& host_buffer = pass.GetTransientsBuffer();
+  using VT = SolidFillVertexShader::PerVertexData;
+
+  Scalar radius = radius_;
+  const Point& center = center_;
+  std::shared_ptr<Tessellator> t = renderer.GetTessellator();
+  CircleTessellator tessellator(t, entity.GetTransform(), radius_);
+  size_t count = tessellator.GetCircleVertexCount();
+  auto vertex_buffer =
+      host_buffer.Emplace(count * sizeof(VT), alignof(VT),
+                          [&tessellator, &center, radius](uint8_t* buffer) {
+                            auto vertices = reinterpret_cast<VT*>(buffer);
+                            tessellator.GenerateCircleTriangleStrip(
+                                [&vertices](const Point& p) {  //
+                                  *vertices++ = {
+                                      .position = p,
+                                  };
+                                },
+                                center, radius);
+                          });
+
+  return GeometryResult{
+      .type = PrimitiveType::kTriangleStrip,
+      .vertex_buffer =
+          {
+              .vertex_buffer = vertex_buffer,
+              .vertex_count = count,
+              .index_type = IndexType::kNone,
+          },
+      .transform = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransform(),
+      .prevent_overdraw = false,
+  };
+}
+
+// |Geometry|
+GeometryResult EllipseGeometry::GetPositionUVBuffer(
+    Rect texture_coverage,
+    Matrix effect_transform,
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) const {
+  auto& host_buffer = pass.GetTransientsBuffer();
+  using VT = TextureFillVertexShader::PerVertexData;
+  auto uv_transform =
+      texture_coverage.GetNormalizingTransform() * effect_transform;
+
+  Scalar radius = radius_;
+  const Point& center = center_;
+  std::shared_ptr<Tessellator> t = renderer.GetTessellator();
```

Comment:
here and elsewhere, I would prefer a longer identifer. Maybe just `tess`?

Reply:
I struggled with that because circle_tessellator is just so long for the other one that gets used more often. Also, you can't transfer a shared_ptr in a method call, as in `tessellator(renderer.GetTessllator(), ...)` because I am asking for a reference to a shared_ptr to avoid incref decref overhead. Perhaps we should look at returning `shared_ptr&` refs from these methods?

In the meantime, I'll adjust the names and take the reformat hit...